BinarySearchTree.playground/Pages/Final Code.xcplaygroundpage/Contents.swift

```diff
@@ -92,35 +91,33 @@ class BinarySearchTree<T: Comparable & CustomStringConvertible>: CustomStringCon
      *  either the biggest child on the left or the smallest child on the right.
      */
     private func remove(_ value: T, fromParent node: BinaryNode<T>?) -> BinaryNode<T>? {
-        
         guard let parent = node else {
             return nil
         }
         
-        if value < parent.data {
+        switch value {
```

Comment:
LMK If this logic is the same or if you think the switch statement makes it more clear